x-pack/legacy/plugins/ml/public/application/util/object_utils.ts

```diff
@@ -11,5 +11,9 @@ export const getNestedProperty = (
   accessor: string,
   defaultValue?: any
 ) => {
-  return accessor.split('.').reduce((o, i) => o?.[i], obj) || defaultValue;
+  const value = accessor.split('.').reduce((o, i) => o?.[i], obj);
+
+  if (value === undefined) return defaultValue;
+
+  return value;
```

Comment:
Would be good to add the cases this fixes to `object_utils.test.ts`. It would not be the exact same as it would include `null` but could `??` instead of `||` solve this too?

Reply:
Good point on adding the tests, @walterra 👍 
As far as using the nullish coalescing operator (??) - it would work but has pretty limited support right now because it's so new. Might be better to update once it's more well supported.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
`babel` would take care of that for us to transpile `??` to supported JS. My suggestion was just a nit, we're on the safe side with your code and the strict check against `undefined`.